components/table/stories/template.js

```diff
@@ -204,65 +283,76 @@ export const Template = ({
 		id=${ifDefined(id)}
 		role=${ifDefined(useCheckboxCell ? "grid" : useDivs ? "table" : undefined)}
 		aria-multiselectable=${ifDefined(useCheckboxCell ? "true" : undefined)}
-		style="max-width: 800px;"
 	>
 		<${theadTag}
 			class="${rootClass}-head"
 			role=${ifDefined(useDivs ? "rowgroup" : undefined)}
 		>
 			<${rowTag}
 				role=${ifDefined(useDivs ? "row" : undefined)}
+				class="${rootClass}-headRow"
 			>
 				${when(useCheckboxCell, () => html`
 					<${thTag}
 						class="spectrum-Table-headCell spectrum-Table-checkboxCell"
 						role=${ifDefined(useDivs ? "columnheader" : undefined)}
 					>
-						${Checkbox({
-							size,
-							isEmphasized: isEmphasized,
-							isChecked: false,
-							isIndeterminate: true,
-							customClasses: [`${rootClass}-checkbox`],
+						${when(selectionMode === "multiple", () => html`
+							${Checkbox({
+								size: "m",
+								isEmphasized: isEmphasized,
+								isChecked: false,
+								isIndeterminate: true,
+								customClasses: [`${rootClass}-checkbox`],
 						}, context)}
-					</${thTag}>`
+						`)}
+					</${thTag}>`,
 				)}
 				<${thTag}
-					class="${rootClass}-headCell is-sortable is-sorted-desc"
+					class=${classMap({
+						[`${rootClass}-headCell`]: true,
+						["is-sortable"]: isSortable,
+						["is-sorted-asc"]: sortIcon === "SortUp",
+						["is-sorted-desc"]: sortIcon === "SortDown",
+						[`${rootClass}-menuButton`]: hasMenu,
+					})}
 					role=${ifDefined(useDivs ? "columnheader" : undefined)}
-					aria-sort="descending"
-					tabindex="0"
+					aria-sort=${ifDefined(isSortable ? ariaSortValue : undefined)}
 				>
-					${Icon({
-						iconName: "ArrowDown100",
-						setName: "ui",
-						size,
-						customClasses: [`${rootClass}-sortedIcon`],
-					}, context)}<span class="${rootClass}-columnTitle">Column title</span>${
-					Icon({
-						iconName: "ChevronDown100",
-						setName: "ui",
-						size,
-						customClasses: [`${rootClass}-menuIcon`],
-					}, context)}
+					${when(isSortable,
+						() => Button({
+							size: "m",
+							iconName: sortIcon,
+							iconSet: "workflow",
+							label: "Column title",
+							customClasses: [`${rootClass}-tableButton`],
+						}, context),
+						() => when(hasMenu,
+							() => Button({
+								size: "m",
+								iconName: "SortUp",
+								iconSet: "workflow",
+								label: "Column title",
+								trailingIcon: "Chevron100",
+								trailingIconSet: "ui",
+								customClasses: [`${rootClass}-tableButton`],
+							}, context),
```

Comment:
I have some questions about this too!

* I don't see `trailingIcon` (or `trailingIconSet`) defined as an arg anywhere in the button template, and removing it isn't doing much that I can tell, do we need it?
* `!isSortable && hasMenu` shows the sort up icon, and `isSortable && hasMenu` shows the sort icon, is that expected?
![image](https://github.com/user-attachments/assets/d62fe1f0-820f-4313-813f-907649c379de)
![image](https://github.com/user-attachments/assets/d6e7a35f-5f88-46c3-ac07-1fb16e1268eb)